### PR TITLE
[TASK] Enhance PSR-14 EventDispatcher documentation

### PR DESCRIPTION
**Pull request by garvinhicking**
This acts as a pre-patch for a distinct section about overriding events.

Releases: main, 12.4

### CODE REVIEW

File: Documentation/ApiOverview/Events/EventDispatcher/Index.rst [hunk 1]
```diff
@@ -220,24 +231,25 @@ The PHP attribute :php:`\TYPO3\CMS\Core\Attribute\AsEventListener` supports the
...
-    and orderings can be build upon the identifier. It this property is not
-    explicitly defined, the service name is used instead.
+    A unique identifier should be declared which identifies the event listener,
+    and orderings can be build upon the identifier. If this property is not
+    explicitly defined, the service name is automatically used instead.
 
 :php:`before`
     This property allows a custom sorting of registered listeners. The listener
-    is then dispatched before the given listener. The value is the identifier of
+    is then dispatched before the given listener. The values are the identifier(s) of
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Events/EventDispatcher/Index.rst:240]: This may be confusing to the reader: Does an event listener have multiple identifiers? That's how the last sentence can be read.

Comment (garvinhicking) [reply Documentation/ApiOverview/Events/EventDispatcher/Index.rst:240]: This came due to your suggestion that multiple identifiers are allowed, maybe rephrase this to:

```
is then dispatched before the given listener. The value is the identifier of another event listeners. Also multiple event listeners can be entered here, separated by a comma.
```

?

File: Documentation/ApiOverview/Events/EventDispatcher/Index.rst [hunk 2]
```diff
@@ -220,24 +231,25 @@ The PHP attribute :php:`\TYPO3\CMS\Core\Attribute\AsEventListener` supports the
...
     another event listeners.
 
 :php:`after`
     This property allows a custom sorting of registered listeners. The listener
-    is then dispatched after the given listener. The value is the identifier of
+    is then dispatched after the given listener. The values are the identifier(s) of
```
[the remark this answers is not in the document]
Comment (garvinhicking) [reply Documentation/ApiOverview/Events/EventDispatcher/Index.rst:245]: see above

File: Documentation/ApiOverview/Events/EventDispatcher/Index.rst (hunk 1 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/Events/EventDispatcher/Index.rst:240]: Yup, fine with that.
Here is the changelog with such an example:
https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/10.0/Feature-88770-PSR-14BasedEventDispatcher.html#impact

Comment (garvinhicking) [reply Documentation/ApiOverview/Events/EventDispatcher/Index.rst]: Changed with different wording clarifying this

File: Documentation/ApiOverview/Events/EventDispatcher/Index.rst [hunk 3]
```diff
@@ -286,10 +300,16 @@ The tag name :yaml:`event.listener` identifies that a listener should be registe
 
 The custom PHP class :php:`\MyVendor\MyExtension\EventListener\NullMailer`
 serves as the listener whose :php:`handleEvent()` method is called, once the
-:yaml:`event` is dispatched. The :yaml:`identifier` is a common name, so
+:yaml:`event` is dispatched. The name of the listened event is specified as
+a typed argument to that dispatch method.
+:php`handleEvent(\TYPO3\CMS\Core\Mail\Event\AfterMailerInitializationEvent $event)` will
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Events/EventDispatcher/Index.rst:305]:
```suggestion
:php:`handleEvent(\TYPO3\CMS\Core\Mail\Event\AfterMailerInitializationEvent $event)` will
```

**Review by brotkrueml — DISMISSED** [review]

**Comment by brotkrueml** [comment]
@garvinhicking Please resolve the conflicts.

**Comment by garvinhicking** [comment]
Resolved conflicts.

**Review by brotkrueml — APPROVED** [review]
